Automattic/wp-calypso
# Domain Management i1: remove auto-renew column; standardise date formatting

Related to p1694004892031289/1694003372.798729-slack-C041RHH38NQ and p1693998497077049/1693984219.175589-slack-C04H4NY6STW

## Proposed Changes

* Standardise date formatting
* Remove the auto-renew column from the table

<img width="910" alt="Screenshot 2566-09-07 at 12 15 48" src="https://github.com/Automattic/wp-calypso/assets/6851384/a10061ba-f124-401e-a25f-59b573f82eee">

## Testing Instructions



* Browse `/domains/manage?flags=domains/management` on this branch
* Submit a bulk auto-renew job and check the popup for the domain 
* Confirm the table now looks like the screenshot

## Pre-merge Checklist
